File: src/main/golang/leetcode/editor/cn/[260]Single Number III.go
package lc

//leetcode submit region begin(Prohibit modification and deletion)
func singleNumber(nums []int) []int {
	var seed int
	for _, num := range nums {
		seed ^= num
	}
	//sign := (seed & (seed - 1)) ^ seed
	sign := seed & -seed
	var s0, s1 int
	for _, num := range nums {
		if sign&num > 0 {
			s0 ^= num
		} else {
			s1 ^= num
		}
	}
	return []int{s0, s1}
}

//leetcode submit region end(Prohibit modification and deletion)

File: src/main/golang/leetcode/editor/cn/[260]Single Number III.py
from typing import List


# leetcode submit region begin(Prohibit modification and deletion)
class Solution:
    def singleNumber(self, nums: List[int]) -> List[int]:
        all = 0
        for num in nums:
            all ^= num
        last_bit = all & -all
        fst = snd = 0
        for num in nums:
            if last_bit & num:
                fst ^= num
            else:
                snd ^= num
        return [fst, snd]
# leetcode submit region end(Prohibit modification and deletion)
